### PR TITLE
Forgot and reset password functionality

### PR DESCRIPTION
**Pull request by IharKrasnik**
@anorsich @ezhivitsa please review

### CODE REVIEW

**Comment by IharKrasnik** [comment]
Should be reviewed together with matching koa-api PR
https://github.com/paralect/koa-api-starter/pull/1

File: src/client/pages/forgot-password.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  submitSignin = async (event) => {
+    event.preventDefault();
+    try {
+      this.setState({ isLoading: true, error: null });
+      await forgotPassword({ email: this.state.email });
+      this.setState({ emailSent: true });
+    } catch (error) {
+      this.setState({ error });
+    } finally {
+      this.setState({ isLoading: false });
```
Comment (ezhivitsa) [inline src/client/pages/forgot-password.js]: I think that here we can use less calls of the `setState` function. Each call to the setState function calls render of the component, so we must avoid unnecessary calls.
```javascript
this.setState({ isLoading: true, error: null });
const newState = {};
try {
  await forgotPassword({ email: this.state.email });
  newState.emailSent = true;
} catch (error) {
  newState.error = error;
} finally {
  this.setState({
    ...newState,
    isLoading: false,
  })
}
```

File: src/client/pages/reset-password.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  }
+
+  constructor(...args) {
+    super(...args);
+    this.setPassword = setFormValue('password').bind(this);
+  }
+
+  componentWillMount() {
+    this.token = this.props.token;
+    this.state = {
```
Comment (ezhivitsa) [inline src/client/pages/reset-password.js]: The same as above

File: src/client/pages/reset-password.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+      error: null,
+    };
+  }
+
+  submitSignin = async (event) => {
+    event.preventDefault();
+
+    try {
+      this.setState({ isLoading: true, error: null });
```
Comment (ezhivitsa) [inline src/client/pages/reset-password.js]: See comment above

File: src/client/pages/forgot-password.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+                    {'Check your email for a link to reset your password. If it doesn\'t appear within a few minutes, check your spam folder.'}
+                  </p>
+                </div>
+              ) : ([
+                <h2> Reset Your Password </h2>,
+                <form className="form" onSubmit={this.submitSignin}>
+                  <input
+                    value={this.state.email}
+                    onChange={this.setEmail}
+                    required
```
Comment (ezhivitsa) [inline src/client/pages/forgot-password.js]: I don't like HTML5 validation. Can we use our own validation?

**Comment by ezhivitsa** [comment]
@anorsich Please, review

**Pull request merged**